tensorboard/backend/application.py

```diff
@@ -65,324 +64,338 @@ def standard_tensorboard_wsgi(
+    plugins = [constructor(context) for constructor in plugins]
+    return TensorBoardWSGIApp(logdir, plugins, multiplexer, reload_interval)
 
-  @wrappers.Request.application
-  def _serve_logdir(self, request):
-    """Respond with a JSON object containing this TensorBoard's logdir."""
-    return http_util.Respond(
-        request, {'logdir': self._logdir}, 'application/json')
 
-  @wrappers.Request.application
-  def _serve_plugins_listing(self, request):
-    """Serves an object mapping plugin name to whether it is enabled.
-
-    Args:
-      request: The werkzeug.Request object.
-
-    Returns:
-      A werkzeug.Response object.
-    """
-    return http_util.Respond(
-        request,
-        {plugin.plugin_name: plugin.is_active() for plugin in self._plugins},
-        'application/json')
+class TensorBoardWSGIApp(object):
+    """The TensorBoard application, conforming to WSGI spec."""
+
+    # How many samples to include in sampling API calls by default.
+    DEFAULT_SAMPLE_COUNT = 10
+
+    # NOTE TO MAINTAINERS: An accurate Content-Length MUST be specified on all
+    #                      responses using send_header.
+    protocol_version = 'HTTP/1.1'
+
+    def __init__(self, logdir, plugins, multiplexer, reload_interval):
+        """Constructs the TensorBoard application.
+
+        Args:
+            logdir: the logdir spec that describes where data will be loaded.
+                may be a directory, or comma,separated list of directories,
+                or colons can be used to provide named directories
+            plugins: A list of base_plugin.TBPlugin subclass instances.
+            multiplexer: The EventMultiplexer with TensorBoard data to serve
+            reload_interval: How often (in seconds) to reload the Multiplexer
+
+        Returns:
+            A WSGI application that implements the TensorBoard backend.
+
+        Raises:
+            ValueError: If some plugin has no plugin_name
+            ValueError: If some plugin has an invalid plugin_name (plugin
+                names must only contain [A-Za-z0-9_.-])
+            ValueError: If two plugins have the same plugin_name
+            ValueError: If some plugin handles a route that does not start
+                with a slash
+        """
+        self._logdir = logdir
+        self._plugins = plugins
+        self._multiplexer = multiplexer
+        self.tag = get_tensorboard_tag()
+
+        path_to_run = parse_event_files_spec(self._logdir)
+        if reload_interval:
+            start_reloading_multiplexer(self._multiplexer, path_to_run,
+                                        reload_interval)
+        else:
+            reload_multiplexer(self._multiplexer, path_to_run)
+
+        self.data_applications = {
+            DATA_PREFIX + LOGDIR_ROUTE:
+                self._serve_logdir,
+            # TODO(chizeng): Delete this RPC once we have skylark rules that
+            # obviate
```

Comment:
It seems like the formatter doesn't always do the right thing with comments like this. Any way to fix this?